### PR TITLE
feat: Write documentation for vue integration TG-383

### PR DESCRIPTION
**Pull request by stepan662**

### CODE REVIEW

**Comment by stepan662** [comment]
Add default value through parameter (so it can support expressions) and mention it in documentation

**Comment by stepan662** [comment]
Investigate why intellisence doesn't work with .ts files

File: docs/web/using_with_vue/advanced_translating.mdx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+```html
+<T keyName="translation_key" strategy="NO_WRAP" />
+```
+
+## [`$t`](./api#function-t) function wrapping
+
+[`$t`](./api#function-t) function is a bit more complicated, because we cannot attach our own html element into the dom.
+So in this case the function is returning tolgee custom made [wrapping](../understanding_tolgee_for_web#key-and-parameters-wrapping),
+which is then captured by dom changes observer and that will find suitable parent element, where the translation gets
```
Comment (JanCizmar) [inline docs/web/using_with_vue/advanced_translating.mdx:24]: dom changes observer -> [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver)

File: docs/web/using_with_vue/advanced_translating.mdx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+registred and so in-context localization can work there. In production mode, this whole logic can be ommited and [`$t`](./api#function-t) function
```
Comment (JanCizmar) [inline docs/web/using_with_vue/advanced_translating.mdx:25]: can be ommited -> is omitted

File: docs/web/using_with_vue/advanced_translating.mdx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+returns only the translation string.
```
Comment (JanCizmar) [inline docs/web/using_with_vue/advanced_translating.mdx:26]: translation string -> translated string

File: docs/web/using_with_vue/advanced_translating.mdx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+This can be problematic in cases where you put translation outside of dom document, because then the change is not
```
Comment (JanCizmar) [inline docs/web/using_with_vue/advanced_translating.mdx:28]: in the cases

Comment (JanCizmar) [reply docs/web/using_with_vue/advanced_translating.mdx:28]: outside of [targetElement](/docs/web/configuration#targetelement)

File: docs/web/using_with_vue/api.mdx [hunk 5]
```diff
@@ -0,0 +1,117 @@
+---
+id: api
+title: API
+sidebar_label: API
+slug: /web/using_with_vue/api
+---
+
+## TolgeeProvider
+Provides tolgee instance. Use in root of your application.
```
Comment (JanCizmar) [inline docs/web/using_with_vue/api.mdx:9]: Tolgee context

File: docs/web/using_with_vue/api.mdx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+```
+
+```html
+<TolgeeProvider config="...">
+  ... content ...
+</TolgeeProvider>
+```
+
+#### Prop `config`
+[`TolgeeConfiguration`](../configuration) object, which is passed to tolgee instance.
```
Comment (JanCizmar) [inline docs/web/using_with_vue/api.mdx:22]: Tolgee

File: docs/web/using_with_vue/api.mdx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+#### Prop `loadingFallback?` 
+`String | JSX.Element` - it is rendered when tolgee is loading translations instead of provided content.
```
Comment (JanCizmar) [inline docs/web/using_with_vue/api.mdx:25]: Tolgee

File: docs/web/using_with_vue/api.mdx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+You can pass custom loading element when using Vue with JSX.
+
+#### Slot `fallback`
+Alternative for `loadingFallback` use when you are using Vue with templates:
```
Comment (JanCizmar) [inline docs/web/using_with_vue/api.mdx:29]: usage?

**Review by JanCizmar — CHANGES_REQUESTED** [review]

File: docs/web/using_with_vue/advanced_translating.mdx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+picked up by the dom observer. You can do that with [`noWrap`](./api#parameter-nowrap) parameter.
```
Comment (marketachalupnikova) [inline docs/web/using_with_vue/advanced_translating.mdx:29]: DOM

File: docs/web/using_with_vue/api.mdx [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+`String` - translation key.
+
+#### Parameter `parameters?`
+`Record<string, string>` - variable parameters, which can be used in translation value 
+(read more about [ICU message format](../../icu_message_format)).
+
+#### Parameter `noWrap?`
+`Boolean` (default: `false`)
+ - `false` - in development mode translation will be [wrapped](../understanding_tolgee_for_web#key-and-parameters-wrapping)
+ - `true` - use when wrapping in dev mode causes problems, or you pass write it outside the dom
```
Comment (marketachalupnikova) [inline docs/web/using_with_vue/api.mdx:102]: DOM

File: docs/web/using_with_vue/api.mdx [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+ - `false` - in development mode translation will be [wrapped](../understanding_tolgee_for_web#key-and-parameters-wrapping)
+ - `true` - use when wrapping in dev mode causes problems, or you pass write it outside the dom 
+    (for example `document.title`), in this case in-context translation won't work
+
+#### Parameter `defaultValue?`
+`String` - It will be rendered if there is no translation for this key (in selected language nor base language).
+If you won't provide this value, `key` will be rendered instead.
+
+### Property `tolgeeLanguage`
+`String` - Reactive property, which contains current tolgee language. Use it for reading or setting current language.
```
Comment (marketachalupnikova) [inline docs/web/using_with_vue/api.mdx:110]: Tolgee

File: docs/web/using_with_vue/translating.mdx [hunk 12]
```diff
@@ -0,0 +1,61 @@
+---
+id: translating
+title: Translating
+sidebar_label: Translating
+slug: /web/using_with_vue/translating
+---
+
+There are two ways how to translate with tolgee.
```
Comment (marketachalupnikova) [inline docs/web/using_with_vue/translating.mdx:8]: Tolgee

**Comment by stepan662** [comment]
Add page Preparing for production (as in svelte docs)

File: docs/web/using_with_vue/api.mdx (hunk 8 above)
Comment (stepan662) [reply docs/web/using_with_vue/api.mdx:29]: I think it's explainded above - in `loadingFallback`

**Pull request merged**